projectgenerator/gui/export.py

```
@@ -21,24 +21,75 @@
 import os.path
+import re
 
-from projectgenerator.gui.options import OptionsDialog, CompletionLineEdit
+from projectgenerator.gui.options import OptionsDialog, ModelListView
 from projectgenerator.gui.multiple_models import MultipleModelsDialog
 from projectgenerator.libili2db.iliexporter import JavaNotFoundError
 from projectgenerator.libili2db.ilicache import IliCache, ModelCompleterDelegate
 from projectgenerator.utils.qt_utils import make_save_file_selector, Validators, \
     make_file_selector, FileValidator, NonEmptyStringValidator, make_folder_selector, OverrideCursor
 from qgis.PyQt.QtGui import QColor, QDesktopServices, QFont, QValidator
 from qgis.PyQt.QtWidgets import QDialog, QDialogButtonBox, QApplication, QCompleter, QMessageBox
-from qgis.PyQt.QtCore import QCoreApplication, QSettings, Qt, QLocale
+from qgis.PyQt.QtCore import QCoreApplication, QSettings, Qt, QLocale, QStringListModel, QTimer
 from qgis.core import QgsProject
 from qgis.gui import QgsGui
 from ..utils import get_ui_class
 from ..libili2db import iliexporter, ili2dbconfig
+from ..libqgsprojectgen.dbconnector import pg_connector, gpkg_connector
 
 DIALOG_UI = get_ui_class('export.ui')
 
 
+class ExportModels(QStringListModel):
+
+    def __init__(self, tool_name, uri, schema=None):
+        super().__init__()
+
+        modelnames = list()
+        try:
+            if tool_name == 'ili2gpkg':
+                self._db_connector = gpkg_connector.GPKGConnector(uri, None)
+            else:
+                self._db_connector = pg_connector.PGConnector(uri, schema)
+
+            if self._db_connector.db_or_schema_exists():
+                db_models = self._db_connector.get_models()
+                for db_model in db_models:
+                    regex = re.compile(r'(?:\{[^\}]*\}|\s)')
+                    for modelname in regex.split(db_model['modelname']):
+                        if modelname:
+                            modelnames.append(modelname.strip())
+        except:
```

Can you catch specific errors here? Catch-all exceptions should be avoided.